.happy/terraform/modules/ecs-stack/main.tf

```diff
@@ -0,0 +1,192 @@
+# This deploys an Aspen stack.
+#
+
+data aws_secretsmanager_secret_version config {
+  secret_id = var.happy_config_secret
+}
+
+locals {
+  secret = jsondecode(data.aws_secretsmanager_secret_version.config.secret_string)
+  alb_key = var.require_okta ? "private_albs" : "public_albs"
+
+  custom_stack_name     = var.stack_name
+  image_tag             = var.image_tag
+  priority              = var.priority
+  deployment_stage      = var.deployment_stage
+  remote_dev_prefix     = var.stack_prefix
+  wait_for_steady_state = var.wait_for_steady_state
+
+  migration_cmd         = ["make", "run-db-migrations"]
+  deletion_cmd          = ["make", "db-drop"]
+  backend_cmd           = []
+  frontend_cmd          = ["npm", "run", "serve"]
+  nextstrain_cmd        = ""
+  data_load_path        = length(var.sql_import_file) > 0 ? "${local.secret["s3_buckets"]["aspen"]["name"]}/${var.sql_import_file}" : ""
+
+  vpc_id                = local.secret["vpc_id"]
+  subnets               = local.secret["private_subnets"]
+  security_groups       = local.secret["security_groups"]
+  zone                  = local.secret["zone_id"]
+  cluster               = local.secret["cluster_arn"]
+  frontend_image_repo   = local.secret["ecrs"]["frontend"]["url"]
+  backend_image_repo    = local.secret["ecrs"]["backend"]["url"]
+  nextstrain_image_repo = local.secret["ecrs"]["nextstrain"]["url"]
+  nextstrain_error_repo = local.secret["ecrs"]["nextstrain-errorhandler"]["url"]
+  batch_role_arn        = local.secret["batch_queues"]["nextstrain"]["role_arn"]
+  job_queue_arn         = local.secret["batch_queues"]["nextstrain"]["queue_arn"]
+  external_dns          = local.secret["external_zone_name"]
+  internal_dns          = local.secret["internal_zone_name"]
+
+  frontend_listener_arn = try(local.secret[local.alb_key]["frontend"]["listener_arn"], "")
+  backend_listener_arn  = try(local.secret[local.alb_key]["backend"]["listener_arn"], "")
+  frontend_alb_zone     = try(local.secret[local.alb_key]["frontend"]["zone_id"], "")
+  backend_alb_zone      = try(local.secret[local.alb_key]["backend"]["zone_id"], "")
+  frontend_alb_dns      = try(local.secret[local.alb_key]["frontend"]["dns_name"], "")
+  backend_alb_dns       = try(local.secret[local.alb_key]["backend"]["dns_name"], "")
```

Comment:
why do these lines have try() statements? Is this because we need to wait for the alb to be setup?

Reply:
These values will / won't be set for remote dev vs staging, so the try() lets us keep the same interface for both sets of environments, even though those alb's won't always be present.

Reply:
Actually I don't think we need these try() calls